Survey/Survey/Sources/Data/Repositories/LoginRepository.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  LoginRepository.swift
+//  Survey
+//
+//  Created by Taher on 4/11/22.
+//
+
+import Combine
+import Foundation
+
+final class LoginRepository: LoginRepositoryProtocol {
+
+    let network: LoginNetworkAPI
+
+    init(network: LoginNetworkAPI) {
+        self.network = network
+    }
+
+    func getLoginToken(email: String, password: String) -> AnyPublisher<LoginTokenApi, Error> {
+        network.performRequest(
+            APIRequestConfigurations.login(email: email, password: password),
+            for: LoginTokenApi.self,
+            with: URLSession.shared
+        )
+    }
+}
```

Comment:
This can drop `with: URLSession.shared` from function call if the `LoginNetworkAPI` already has it as a dependency!

Reply:
updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/43/commits/08e91693e844c6ce389503dced572cfcc5f17cea)